src/utils.cpp

```diff
@@ -454,4 +454,19 @@ void cat_file(const char *filename, size_t max_bytes, std::ostream &out)
   }
 }
 
+std::string str_join(const std::vector<std::string> &list, const std::string &delim)
+{
+  std::string str;
+  int i = 0;
```

Comment:
The `i++` looks kind of strange. Can we do a `bool first=true` flag type thing or just use a regular `for (int i = 0; ...` loop?